File: src/main/java/com/hjj/apiserver/domain/SavingPK.java
package com.hjj.apiserver.domain;

import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import java.io.Serializable;

@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@NoArgsConstructor
public class SavingPK implements Serializable {


    private static final long serialVersionUID = -4052438448409315836L;

    @EqualsAndHashCode.Include
    private String bank;    //finCoNo
    @EqualsAndHashCode.Include
    private String finPrdtCd;

    public SavingPK(String finCoNo, String finPrdtCd) {
        this.bank = finCoNo;
        this.finPrdtCd = finPrdtCd;
    }
}

File: src/main/kotlin/com/hjj/apiserver/domain/saving/SavingPK.kt
package com.hjj.apiserver.domain.saving

import jakarta.persistence.Embeddable
import java.io.Serializable

@Embeddable
data class SavingPK(
    var bank: String,
    var finPrdtCd: String,
) : Serializable {
    companion object {
        private const val serialVersionUID: Long = 5473249419463926971L
    }
}
